homeassistant/components/wled/light.py

```diff
@@ -313,7 +314,7 @@ async def async_turn_off(self, **kwargs: Any) -> None:
     @wled_exception_handler
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on the light."""
-        data = {ATTR_ON: True, ATTR_SEGMENT_ID: self._segment}
+        data: dict[str, Any] = {ATTR_ON: True, ATTR_SEGMENT_ID: self._segment}
```

Comment:
`Any` should not be used here, as it is clear which types it could contain.

Reply:
In this method, `data` would have type something like`dict[str, bool | int | tuple[int, ...]]` for the various data types added into it. As there is computation performed on the values, it generates a bunch of mypy errrors like:

```
homeassistant/components/wled/light.py:360: error: Unsupported operand types for + ("bool" and "Tuple[Any]")  [operator]
```

Options:
   1. Use `Any`.
   1. Don't use a 'misc' dict like this at all, store/compute all the values separately rather than piling everything into `data`.
   1. Use a TypedDict.
   1. ?

If you'd rather not use `Any`, I think option 2 is cleanest. Thoughts?

Reply:
I personally would add something like `dict[str, bool | int | tuple[int, ...]]`

Reply:
Apologies, I wasn't clear -- if you set the type to that, then you _still_ get the errors shown above, as it doesn't make sense to `+` something to a tuple, whereas it does for an int. Having several different value types in one dict is difficult when the method later requires arithmetic/modification of those values. 

Cleanest options would appear to be `Any`, replacing `data` entirely with individual data types, or abandon this PR and leave wled untyped.

Reply:
ok, that is good reasoning for Any indeed.
I'm fine with leaving it untyped as well. 🤷